sbol2/partshop.py

```diff
@@ -323,3 +326,75 @@ def attachFile(self, top_level_uri, filepath):
         msg = 'HTTP Error code {} trying to attach file.'
         msg = msg.format(response.status_code)
         raise SBOLError(msg, SBOLErrorCode.SBOL_ERROR_BAD_HTTP_REQUEST)
+
+    def _make_search_item(self, item:dict) -> Identified:
+        obj = Identified()
+        obj.identity = item['uri']
+        obj.displayId = item['displayId']
+        obj.name = item['name']
+        obj.description = item['description']
+        obj.version = item['version']
+        return obj
+
+    def _search(self, url:str) -> List[Identified]:
```

Comment:
Would it be helpful to implement a generic method for structuring get requests?

There is a method `http_get_request` in libSBOL that does this:
https://github.com/SynBioDex/libSBOL/blob/master/source/partshop.cpp#L1060

It was added to the library relatively late, so many PartShop methods don't actually take advantage of it.  An advantage of a generic method is that it enables a single code change to update of all the methods if, for example, Synbiohub changes its http headers.

Reply:
Yes, absolutely. I've been contemplating that. I think we want that as we [implement more part shop endpoints](#180)